src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -245,9 +247,15 @@ private Delegate GetGetterCore(DataViewRow input, int iinfo, out Action disposer
             disposer = null;
             input.Schema.TryGetColumnIndex(_columns[iinfo].InputColumnName, out int srcCol);
             var srcType = input.Schema[srcCol].Type;
-            if (!(srcType is VectorDataViewType vectorType))
-                return ComposeGetterOne(input, iinfo, srcCol, srcType);
-            return ComposeGetterVec(input, iinfo, srcCol, vectorType);
+            if (GetVersionInfo().VerWrittenCur == 0x00010002)
```

Comment:
>GetVersionInfo() [](start = 16, length = 16)

This calls the method `GetVersionInfo`, which always returns the same thing (currently 0x00010002, after your change 0x00010003).
The correct way to do what you are trying to achieve, is to add a `_version` field to the transformer. This field would be initialized according to `ctx.Header.ModelVerWritten` in the constructor from file, and to 0x00010003 in the other constructors.

Reply:
Added versioning + test